docs/reference/release-notes/8.16.4.asciidoc

```diff
@@ -39,4 +39,10 @@ Snapshot/Restore::
 Ingest Node::
 * Improve memory aspects of enrich cache {es-pull}120256[#120256] (issues: {es-issue}96050[#96050], {es-issue}120021[#120021])
 
+[discrete]
+[[known-issues-8.16.4]]
+=== Known issues
 
+ES|QL::
+
+* `VALUES` aggregate function can run for a long, long time when collecting many, many groups. Hundreds of thousands of groups can run on one thread for many minutes and millions of groups run for days. It is O(n^2^) with the number of groups. These will not respond to the tasks cancellation API the whole time. Fixed by {es-pull}#123073[#123073] and available in 8.16.5, 8.17.3, 8.18.0, and all releases after that.
```

Comment:
```suggestion
Here's the revised copyedited version:

* The `VALUES` aggregate function can run for an extremely long time when collecting many groups.  Processing hundreds of thousands of groups may take many minutes on a single thread, while processing millions of groups can take days. The function has O(n^2) complexity with respect to the number of groups. During execution, these operations will not respond to the tasks cancellation API. This issue has been fixed by {es-pull}#123073[#123073] and the fix is available in versions 8.16.5, 8.17.3, 8.18.0, and all releases after that.
```

Reply:
nitty copyediting 🤓

Reply:
I think that makes it worse, but I can use that if you'd prefer.

Reply:
It sort of sucks the voice out of it.